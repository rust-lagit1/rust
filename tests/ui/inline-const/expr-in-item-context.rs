// reported in https://github.com/rust-lang/rust/issues/128338

// This had an overly-terse diagnostic, because we recognized this is where an item should be, but
// but didn't recognize an inline const in this place is probably trying to be an "anon const item".
// Those are written like this:
const _: () = {};
// const _ are often used as compile-time assertions that don't conflict with other const items

const { assert!(size_of::<u32>() <= size_of::<usize>()) };
//~^ expected item, found keyword
//~| to evaluate a const expression, use an anonymous const

fn main() {
    const { assert!(size_of::<u32>() <= size_of::<usize>()) };
}
