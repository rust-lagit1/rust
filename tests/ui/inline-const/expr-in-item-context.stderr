error: expected item, found keyword `const`
  --> $DIR/expr-in-item-context.rs:9:1
   |
LL | const { assert!(size_of::<u32>() <= size_of::<usize>()) };
   | ^^^^^
   |
help: to evaluate a const expression, use an anonymous const
   |
LL | const _: () = const { assert!(size_of::<u32>() <= size_of::<usize>()) };
   | +++++++++++++

error: aborting due to 1 previous error

